### PR TITLE
Add MemoryFormat for clone

### PR DESCRIPTION
**Pull request by ailzhang**
This should fix the breakage once https://github.com/pytorch/pytorch/pull/26904 lands. 
I will trigger the test again once after it lands.

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -807,7 +807,9 @@ at::Tensor& AtenXlaType::clamp_min_(at::Tensor& self, at::Scalar min) {
   return self;
 }
 
-at::Tensor AtenXlaType::clone(const at::Tensor& self) {
+at::Tensor AtenXlaType::clone(
+    const at::Tensor& self,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: ```c10::optional<c10::MemoryFormat> /* optional_memory_format */```

**Comment by VitalyFedyunin** [comment]
https://github.com/pytorch/pytorch/pull/27106 is going to be used instead of https://github.com/pytorch/pytorch/pull/26904

**Comment by ailzhang** [comment]
Restarted CI before merging.

File: torch_xla/csrc/aten_xla_type.h [hunk 2]
```diff
@@ -288,9 +288,8 @@ class AtenXlaType {
 
   static at::Tensor& clamp_min_(at::Tensor& self, at::Scalar min);
 
-  static at::Tensor clone(
-      const at::Tensor& self,
-      c10::optional<c10::MemoryFormat> optional_memory_format);
+  static at::Tensor clone(const at::Tensor& self,
+                          c10::optional<at::MemoryFormat> memory_format);
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.h]: They don't match the namespace here now

**Pull request merged**